File: backend/fief/schemas/tenant.py
from fief.schemas.generics import BaseModel, CreatedUpdatedAt, UUIDSchema


class TenantCreate(BaseModel):
    name: str
    registration_allowed: bool = True


class TenantUpdate(BaseModel):
    name: str | None
    registration_allowed: bool | None


class BaseTenant(UUIDSchema, CreatedUpdatedAt):
    name: str
    default: bool
    slug: str
    registration_allowed: bool


class Tenant(BaseTenant):
    pass


class TenantEmbedded(BaseTenant):
    pass


class TenantEmailContext(BaseTenant):
    pass

File: admin-frontend/src/schemas/tenant.ts
import { CreatedUpdatedAt, PaginationParameters, UUIDSchema } from './generics';

export interface TenantCreate {
  name: string;
  registration_allowed: boolean;
}

export interface TenantUpdate {
  name?: string;
  registration_allowed?: boolean;
}

interface BaseTenant extends UUIDSchema, CreatedUpdatedAt {
  name: string;
  default: boolean;
  slug: string;
  registration_allowed: boolean;
}

export interface Tenant extends BaseTenant {
}

export interface TenantEmbedded extends BaseTenant {
}

export interface TenantListParameters extends PaginationParameters {
  query?: string;
}
